lib/foodcritic/command_line.rb

```diff
@@ -77,20 +81,108 @@ def initialize(args)
         end
         opts.on('-X', '--exclude PATH',
                 'Exclude path(s) from being linted.') do |e|
-          options[:exclude_paths] << e
+          @options[:exclude_paths] << e
+        end
+        opts.on('-F', '--config PATH',
+                'Path to foodcritic.yml file to use.') do |f|
+          @options[:config] = f
         end
       end
+
       # -v is not implemented but OptionParser gives the Foodcritic's version
       # if that flag is passed
       if args.include? '-v'
         help
       else
         begin
+        cbs = @parser.permute(args)
+        rescue OptionParser::InvalidOption => e
+          puts e.message
+        end
+
+        get_config(cbs)
+
+	# Tags handled differently to allow for proper precidence between config files, cookobook files, and command line.
+        @options[:tags]= @fc_config['tags'].flatten.uniq if @fc_config['tags']
```

Comment:
You should be able to do a sane single merge:

```
irb(main):001:0> f = {'a' => 'x', 'b' => 'y', 'c' => 'z'}
=> {"a"=>"x", "b"=>"y", "c"=>"z"}
irb(main):002:0> g = {'p' => 'l', 'q' => 'm', 'r' => 'n'}
=> {"p"=>"l", "q"=>"m", "r"=>"n"}
irb(main):003:0> f.merge(g)
=> {"a"=>"x", "b"=>"y", "c"=>"z", "p"=>"l", "q"=>"m", "r"=>"n"}
irb(main):004:0> f.merge!(g)
=> {"a"=>"x", "b"=>"y", "c"=>"z", "p"=>"l", "q"=>"m", "r"=>"n"}
irb(main):005:0> 
```

Regardless of whether there is overlap. Then walk through and flatten what needs to be flattened.